crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -1056,13 +1057,13 @@ where
     P2: ExprParser + 'a,
 {
     if precedence == Precedence::Highest {
-        if is_type_expression {
-            type_expression_term(expr_parser).boxed().labelled(ParsingRuleLabel::Term)
+        let expr_parser = if is_type_expression {
+            type_expression_term(expr_parser).boxed()
         } else {
-            term(expr_parser, expr_no_constructors, allow_constructors)
-                .boxed()
-                .labelled(ParsingRuleLabel::Term)
-        }
+            term(expr_parser, expr_no_constructors, allow_constructors).boxed()
+        };
+
+        expr_parser.clone().or(parenthesized(expr_parser)).boxed().labelled(ParsingRuleLabel::Term)
```

Comment:
The `type_expression_atom` parser should already support parenthesized type expressions, do you know why we need this extra check?